src/consensus/proposer.rs

```diff
@@ -183,6 +183,13 @@ impl Proposer for ShardProposer {
             self.publish_new_shard_chunk(&chunk.clone()).await;
             self.engine.commit_shard_chunk(&chunk);
             self.proposed_chunks.remove(&value);
+        } else {
+            error!(
+                height = commits.height.unwrap().to_string(),
+                round = commits.round,
+                shard_hash = hex::encode(value.hash),
+                "Unable to find proposal for decided value"
```

Comment:
I think we should panic here. We cannot progress after this, because this node will not be able to propose or validate any subsequent height correctly. It would not have committed a chunk to disk.